diff --git a/bbhash.h b/bbhash.h
index 64075f3..47cf13a 100755
--- a/bbhash.h
+++ b/bbhash.h
@@ -12,6 +12,7 @@
 #include <memory>
 #include <algorithm>
 #include <array>
+#include <cstdint>
 #include <unordered_map>
 #include <vector>
 #include <string>
diff --git a/bm.h b/bm.h
index e4eacd1..04056dd 100755
--- a/bm.h
+++ b/bm.h
@@ -511,7 +511,7 @@ public:
         {
             bvect_ = ii.bvect_;
             if (!buf_)
-                buf_ = bvect_->allocate_tempblock();
+                buf_ = (value_type*) bvect_->blockman_.get_allocator().alloc_bit_block();
             buf_size_ = ii.buf_size_;
             ::memcpy(buf_, ii.buf_, buf_size_ * sizeof(*buf_));
             sorted_ = ii.sorted_;
@@ -522,7 +522,7 @@ public:
         {
             bvect_ = ii.bvect_;
             if (buf_)
-                bvect_->free_tempblock(buf_);
+                bvect_->blockman_.get_allocator().free_bit_block((bm::word_t*) buf_);
             buf_ = ii.buf_; ii.buf_ = 0;
             buf_size_ = ii.buf_size_;
             sorted_ = ii.sorted_;
